src/main/java/com/zaxxer/sparsebits/SparseBitSet.java

```diff
@@ -1035,50 +1050,43 @@ public int previousClearBit(int i)
         }
 
         final long[][][] bits = this.bits;
-        final int aLength = bits.length;
+        final int aSize = bits.length - 1;
 
         int w = i >> SHIFT3;
         int w3 = w & MASK3;
         int w2 = (w >> SHIFT2) & MASK2;
         int w1 = w >> SHIFT1;
-        if (w1 > aLength - 1)
+        if (w1 > aSize)
             return i;
-        w1 = Math.min(w1, aLength - 1);
-        final int w4 = i % LENGTH4;
+        w1 = Math.min(w1, aSize);
+        int w4 = i % LENGTH4;
 
         long word;
         long[][] a2;
         long[] a3;
 
-        final int f3 = w3;
-        final int f2 = w2;
-        final int f1 = w1;
-
         for (; w1 >= 0; --w1)
         {
             if ((a2 = bits[w1]) == null)
-                return (((w1 << SHIFT1) + (w2 << SHIFT2) + w3) << SHIFT3)
-                        + (f1 == w1 ? w4 : LENGTH4 - 1);
+                return (((w1 << SHIFT1) + (w2 << SHIFT2) + w3) << SHIFT3) + w4;
             for (; w2 >= 0; --w2)
             {
                 if ((a3 = a2[w2]) == null)
-                    return (((w1 << SHIFT1) + (w2 << SHIFT2) + w3) << SHIFT3)
-                            + (f2 == w2 ? w4 : LENGTH4 - 1);
+                    return (((w1 << SHIFT1) + (w2 << SHIFT2) + w3) << SHIFT3) + w4;
                 for (; w3 >= 0; --w3)
                 {
                     if ((word = a3[w3]) == 0)
-                        return (((w1 << SHIFT1) + (w2 << SHIFT2) + w3) << SHIFT3)
-                                + (f3 == w3 ? w4 : LENGTH4 - 1);
+                        return (((w1 << SHIFT1) + (w2 << SHIFT2) + w3) << SHIFT3) + w4;
                     for (int bitIdx = w4; bitIdx >= 0; --bitIdx)
                     {
                         if ((word & (1L << bitIdx)) == 0)
                             return (((w1 << SHIFT1) + (w2 << SHIFT2) + w3) << SHIFT3) + bitIdx;
                     }
+                    w4 = LENGTH4_SIZE;
```

Comment:
This is the pertinent change, the rest are just to make the the two methods clearer.